app/jobs/regular/group_smtp_email.rb

```diff
@@ -4,13 +4,48 @@
 
 module Jobs
   class GroupSmtpEmail < ::Jobs::Base
+    include Skippable
+
     sidekiq_options queue: 'critical'
 
     def execute(args)
+      return if quit_email_early?
+
       group = Group.find_by(id: args[:group_id])
       post = Post.find_by(id: args[:post_id])
       email = args[:email]
       cc_addresses = args[:cc_emails]
+      recipient_user = ::UserEmail.find_by(email: email, primary: true)&.user
```

Comment:
It looks like we only need to fetch this record if certain conditions have been met. I think we should extract this to a method so that the DB query doesn't have to be executed upfront. Also we can memoize this in case it is called more than once.

Reply:
I didn't extract this because it's used in the skip logs, but I think I made it better by making a different scope on User to find by the primary email so we aren't loading the UserEmail record just to get the User here